### DISCUSSION TITLE
Shorthand syntax for duplicate validation rules

### DISCUSSION THREAD

**Discussion by jdarkins** [laravel/framework#52165] [Ideas] [answered]
I often find cases where i am writing duplicate rules for form validations. Something along these sort of lines (not real code, just a basic example):

```
     $request->validate([
          'name' => ['required', 'string'],
          'email' => ['required', 'email'],
          'add1' => ['required', 'string'],
          'add2' => ['required', 'string'],
          'add3' => ['required', 'string'],
          'town' => ['required', 'string'],
          'postcode' => ['required', 'string'],
     ])
```

I think it would be great if this could work similar to a match where you can have multiple attributes in a single arm, like so:

```
$request->validate(
'name', 'add1', 'add2', 'add3', 'town', 'postcode' => ['required', 'string'],
'email' => ['required', 'email'],
)
```

I would love to give this a go myself but honestly wouldn't know where to start 😅 

I appreciate it might not even possible / would be a breaking change to move away from passing in an array, I just think it could be a cool idea and couldn't find anyone else talking about it online.

**Comment by henzeb** [comment, marked as answer]
I'd create class constants for these. Or we can add Rule alias feature where you globally define "requiredEmail" => 'required|email'.

**Reply by jdarkins** [reply]
Yeah fair enough

**Comment by ghost** [comment]
@jdarkins this is not valid php code. You forgot a set of []

```
$request->validate([
    'name', 'add1', 'add2', 'add3', 'town', 'postcode' => ['required', 'string'],
    'email' => ['required', 'email'],
])
```

But this is a syntax for match not for defining arrays.

Maybe this is what you wanted (csv keys):

```
$request->validate([
    'name,add1,add2,add3,town,postcode' => ['required', 'string'],
    'email' => ['required', 'email'],
])
```

**Comment by apih** [comment]
Why not write the rules like this?

```php
$rules = [];

foreach (['name', 'add1', 'add2', 'add3', 'town', 'postcode'] as $attribute) {
     $rules[$attribute] = ['required', 'string'];
}

$request->validate($rules + [
     'email' => ['required', 'email'],
]);
```

**Reply by ghost** [reply]
"Because laravel is a wizard that can hold a lot of overhead code to do things in 100 ways."

**Reply by ghost** [reply]
@apih using + is not recommended. array_merge() or array_replace() should be used (taking into account that this is not a list).

See https://stitcher.io/blog/array-merge-vs+
